packages/riipen-ui/src/components/Popover.jsx

```diff
@@ -274,140 +120,140 @@ class Popover extends React.Component {
-  lockParentScroll() {
-    const anchorEl = this.getAnchorEl();
-    const parent = anchorEl.parentElement;
-    const container = getDocument(anchorEl);
+    if (anchorEl) {
+      manageScrollableParentEventListeners(false);
+    }
 
-    this.state.scrollContainer =
-      parent.nodeName === "HTML" &&
-      window.getComputedStyle(parent)["overflow-y"] === "scroll"
-        ? parent
-        : container.body;
+    setPositioningStyle();
 
-    this.state.scrollContainerStyle = this.state.scrollContainer.style.overflow;
-    this.state.scrollContainer.style.overflow = "hidden";
-  }
+    return () => {
+      window.removeEventListener("resize", positionChangeEventHandler);
+      window.removeEventListener("scroll", positionChangeEventHandler);
+      window.removeEventListener("mousedown", handleCloseEvent);
+      window.removeEventListener("keydown", handleCloseEvent);
 
-  resetParentScroll() {
-    this.state.scrollContainer.style.overflow = this.state.scrollContainerStyle;
-  }
+      manageScrollableParentEventListeners(true);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (lockScroll && anchorEl) {
+      const anchorElement = getAnchorEl();
+      const parent = anchorElement.parentElement;
+      const outerNode =
+        parent.nodeName === "HTML" &&
+        window.getComputedStyle(parent)["overflow-y"] === "scroll"
+          ? parent
+          : getDocument(anchorElement).body;
+      setScrollContainer(outerNode);
+      setScrollContainerStyle(outerNode.style.overflow);
+      outerNode.style.overflow = "hidden";
+    } else if (lockScroll && scrollContainer && !anchorEl) {
+      scrollContainer.style.overflow = scrollContainerStyle;
+      setScrollContainer(null);
+      setScrollContainerStyle(null);
+    }
+  }, [anchorEl, lockScroll]);
 
-  updatePosition = () => {
-    if (!this.props.anchorEl) return undefined;
-    return this.setPositioningStyle();
-  };
+  useEffect(() => {
+    manageScrollableParentEventListeners(false);
+  }, [anchorPosition, contentPosition]);
 
-  static contextType = ThemeContext;
-
-  render() {
-    const {
-      anchorEl,
-      classes,
-      children,
-      component,
-      fullWidth,
-      styles,
-      isOpen
-    } = this.props;
-    const theme = this.context;
-    const className = clsx(classes, "popover", { open: isOpen });
-    const Component = component;
-
-    return (
-      <React.Fragment>
-        {anchorEl && isOpen && (
-          <Component
-            style={{ ...styles, ...this.state.contentStyles }}
-            ref={this.contentRef}
-            className={className}
-          >
-            {children}
-          </Component>
-        )}
-        <style jsx>{`
+  useEffect(() => {
+    if (isOpen) {
+      // need to use requestAnimationFrame in order for popover to get proper position on first open
+      window.requestAnimationFrame(setPositioningStyle);
+    }
+  }, [anchorEl, anchorPosition, contentPosition, isOpen]);
```

Comment:
I am not sure if this matches with the previous logic. `isOpen` needs not to be checked whether changes occurred (just want to if that is true). Looks like this logic is also missing `const anchorChange =
      prevProps.anchorEl === null && this.props.anchorEl !== null;`

Reply:
Should be fixed now?